### DISCUSSION TITLE
Livewire 3 Validation

### DISCUSSION THREAD

**Discussion by MotionPhix** [livewire/livewire#5884] [1. Help]
In Livewire 3, the new validation kind of is causing issues with the good old Rule validation from `Illuminate\Validation\Rule` because now it's impossible to do:

```
`#[Rule('email:rfc,dns', 'max:55', Rule::unique(User::class)->ignore(Auth::user()->id))]
 public $email;`
```
  
  How can I make this right?

**Comment by Perer876** [comment]
After inspecting the code, it seems that the older way of defining rules is still available. In fact, Livewire merges together all the rules, allowing you to use both methods. It'll be good if this is mentioned in the docs.

In complex cases you could use this:
```php
namespace App\Livewire;
 
use App\Models\User;
use Illuminate\Validation\Rule;
use Livewire\Attributes\Rule as Rules;
use Livewire\Component;
 
class Foo extends Component
{
    public mixed $email = '';
 
    #[Rules('required')] 
    public mixed $otherProperty = '';
 
    protected function rules(): array
    {
        return [
            'email' => ['email:rfc,dns', 'max:55', Rule::unique(User::class)->ignore(Auth::user()->id)]
        ];
    }
}
```
Remember that if you use both methods, there may be a naming collision because the attributes `Livewire\Attributes\Rule` and the factory `Illuminate\Validation\Rule` have the same name. You could rename one of them or use a fully qualified name.

Also, notice that the `validate` and `validateOnly` methods still accept the rules, but in this case, Livewire doesn't merge the rules defined with `Livewire\Attributes\Rule`.

**Comment by Perer876** [comment]
Please consider that if you define rules both with attributes and using the old method for the same property, Livewire takes precedence over the rules defined by the attributes.

**Comment by joshhanley** [comment]
@MotionPhix yeah @Perer876 is correct, you should use the `rules()` method for this, as using `Rule::exists()...` is an instantiated object which attributes can't accept.

**Comment by MotionPhix** [comment]
That was my worry in the first place to mix the two, but I have used `

> rules()

` way to validate the email only. Thank you for your insight.

**Reply by joshhanley** [reply]
@MotionPhix no worries! 🙂
